sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -92,67 +88,23 @@
   public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptContext)
       throws IOException, InterruptedException {
     Configuration configuration = taskAttemptContext.getConfiguration();
-    ParquetInputSplit split = (ParquetInputSplit)inputSplit;
+    FileSplit split = (FileSplit) inputSplit;
     this.file = split.getPath();
-    long[] rowGroupOffsets = split.getRowGroupOffsets();
-
-    ParquetMetadata footer;
-    List<BlockMetaData> blocks;
 
-    // if task.side.metadata is set, rowGroupOffsets is null
-    if (rowGroupOffsets == null) {
-      // then we need to apply the predicate push down filter
-      footer = readFooter(configuration, file, range(split.getStart(), split.getEnd()));
-      MessageType fileSchema = footer.getFileMetaData().getSchema();
-      FilterCompat.Filter filter = getFilter(configuration);
-      blocks = filterRowGroups(filter, footer.getBlocks(), fileSchema);
-    } else {
```

Comment:
@sunchao, can we keep this code path? I know it has a dead code branch but this code is supposed to contribute back to Parquet. It think it's better to work in the other case too.

Reply:
Sorry for the late response @HyukjinKwon .

Hmm. How likely we are going to keep this code even if move to parquet-mr? the `ParquetInputSplit` this code depends on is already deprecated and will be removed in 2.0. Also, with `rowGroupOffsets` it requires the client (as opposed to tasks) to parse Parquet row group metadata which could get very expensive (which is the motivation that parquet-mr switched to use `FileSplit`).

Reply:
Okie, removal is fine by me.